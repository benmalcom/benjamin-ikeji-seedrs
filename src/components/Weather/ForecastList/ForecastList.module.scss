.container {
  margin: 1.5rem auto;
}

.grid {
  box-shadow: 0 0 2px #808080;
  display: flex;

}

.box {
  background-color: #FFFFFF;
  height: 2rem;
  display: flex;
  align-items: center;
  width: fit-content;
  padding: 0.5rem;
  justify-content: space-between;
  box-shadow: 0 0 1px #808080;
}

.city {
  font-weight: bold;
  font-size: 1.5rem;
}

.cancel {
  margin-left: 1rem;
  font-size: 1rem;
  cursor: pointer;
}

.no-forecast {
  padding: 0.5rem;
  width: 100%;
}
