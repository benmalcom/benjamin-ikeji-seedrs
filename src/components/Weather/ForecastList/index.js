import React from 'react';
import { v1 as uuid } from 'uuid';
import 'slick-carousel/slick/slick.css';
import 'slick-carousel/slick/slick-theme.css';
import styles from './ForecastList.module.scss';

import Forecast from '../Forecast';
import { Container, Grid, Paper, Typography } from '@material-ui/core';
import PropTypes from 'prop-types';
import forecastProptypes from 'utils/prop-types/forecast';

const ForecastList = (props) => {
  const { currentMetric, removeForecast, forecast } = props;

  return <Container className={styles.container}>
    <Grid item xl={12} lg={12} md={12} sm={12} xs={12} className={styles.box}>
      <span className={styles.city}>{forecast.city}</span>
      <span className={styles.cancel} onClick={() => removeForecast(forecast.id)}>x</span>
    </Grid>
    <Grid container item xl={12} lg={12} md={12} sm={12} xs={12} className={styles.grid}>

      {
        forecast.list?.length > 0 ? forecast.list.map(item => <Grid key={uuid()}  item xs={12} lg={4} md={4} sm={6}>
          <Forecast
            currentMetric={currentMetric}
            weatherData={item} />
        </Grid>) : <Grid item xl={12} lg={12} md={12} sm={12} xs={12} className={styles.grid}>
          <Paper className={styles['no-forecast']}>
            <Typography variant="subtitle1" component="h2" color="error">
              {forecast.message}
            </Typography>
          </Paper>
        </Grid>
      }
    </Grid>
  </Container>;
};

ForecastList.propTypes = {
  forecast: PropTypes.shape({
    list: PropTypes.arrayOf(PropTypes.shape(forecastProptypes)).isRequired,
    id: PropTypes.string.isRequired,
    city: PropTypes.string.isRequired,
  }),
  currentMetric: PropTypes.string.isRequired,
  removeForecast: PropTypes.func.isRequired,
};

export default ForecastList;