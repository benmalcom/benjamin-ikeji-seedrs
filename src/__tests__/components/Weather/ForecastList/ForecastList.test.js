import React from 'react';
import { render } from '@testing-library/react';
import ForecastList from 'components/Weather/ForecastList';
import mockWeatherData from 'utils/mock-data/weatherData.json';
import { createTestStore } from 'utils/test/test-utils';
import { Provider } from 'react-redux';

describe('ForecastList', () => {
  const getComponent = props => render(<Provider store={createTestStore()}><ForecastList {...props} /></Provider>);

  it('renders without crash', async () => {
    const forecast = {
      list: mockWeatherData,
      city: 'London',
      id: "uuid123"
    };
    const removeForecast = jest.fn();
    const currentMetric = 'C';
    const component = getComponent({
      forecast,
      removeForecast,
      currentMetric
    });
    expect(component).toMatchSnapshot();
  });
});