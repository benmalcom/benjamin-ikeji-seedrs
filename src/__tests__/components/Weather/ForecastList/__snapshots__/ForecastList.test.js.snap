// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastList renders without crash 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiContainer-root container MuiContainer-maxWidthLg"
      >
        <div
          class="MuiGrid-root box MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-12 MuiGrid-grid-lg-12 MuiGrid-grid-xl-12"
        >
          <span
            class="city"
          >
            London
          </span>
          <span
            class="cancel"
          >
            x
          </span>
        </div>
        <div
          class="MuiGrid-root grid MuiGrid-container MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-12 MuiGrid-grid-lg-12 MuiGrid-grid-xl-12"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4 MuiGrid-grid-lg-4"
          >
            <div
              class="MuiPaper-root MuiCard-root card MuiPaper-elevation1 MuiPaper-rounded"
            >
              <div
                class="MuiCardContent-root"
              >
                <h6
                  class="MuiTypography-root MuiTypography-subtitle2 MuiTypography-colorTextSecondary MuiTypography-alignCenter"
                >
                  24 Apr, 2021
                </h6>
                <img
                  alt="forecast-icon"
                  class="MuiCardMedia-root image MuiCardMedia-media MuiCardMedia-img"
                  src="http://openweathermap.org/img/wn/01n@2x.png"
                />
                <p
                  class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-alignCenter"
                >
                  Clear
                </p>
                <h2
                  class="MuiTypography-root MuiTypography-h4 MuiTypography-gutterBottom MuiTypography-alignCenter"
                >
                  277°C
                </h2>
                <p
                  class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-colorTextSecondary MuiTypography-alignCenter"
                >
                  Feels like 
                  287°C
                </p>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4 MuiGrid-grid-lg-4"
          >
            <div
              class="MuiPaper-root MuiCard-root card MuiPaper-elevation1 MuiPaper-rounded"
            >
              <div
                class="MuiCardContent-root"
              >
                <h6
                  class="MuiTypography-root MuiTypography-subtitle2 MuiTypography-colorTextSecondary MuiTypography-alignCenter"
                >
                  24 Apr, 2021
                </h6>
                <img
                  alt="forecast-icon"
                  class="MuiCardMedia-root image MuiCardMedia-media MuiCardMedia-img"
                  src="http://openweathermap.org/img/wn/01n@2x.png"
                />
                <p
                  class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-alignCenter"
                >
                  Clear
                </p>
                <h2
                  class="MuiTypography-root MuiTypography-h4 MuiTypography-gutterBottom MuiTypography-alignCenter"
                >
                  276°C
                </h2>
                <p
                  class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-colorTextSecondary MuiTypography-alignCenter"
                >
                  Feels like 
                  276°C
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiContainer-root container MuiContainer-maxWidthLg"
    >
      <div
        class="MuiGrid-root box MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-12 MuiGrid-grid-lg-12 MuiGrid-grid-xl-12"
      >
        <span
          class="city"
        >
          London
        </span>
        <span
          class="cancel"
        >
          x
        </span>
      </div>
      <div
        class="MuiGrid-root grid MuiGrid-container MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-12 MuiGrid-grid-lg-12 MuiGrid-grid-xl-12"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4 MuiGrid-grid-lg-4"
        >
          <div
            class="MuiPaper-root MuiCard-root card MuiPaper-elevation1 MuiPaper-rounded"
          >
            <div
              class="MuiCardContent-root"
            >
              <h6
                class="MuiTypography-root MuiTypography-subtitle2 MuiTypography-colorTextSecondary MuiTypography-alignCenter"
              >
                24 Apr, 2021
              </h6>
              <img
                alt="forecast-icon"
                class="MuiCardMedia-root image MuiCardMedia-media MuiCardMedia-img"
                src="http://openweathermap.org/img/wn/01n@2x.png"
              />
              <p
                class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-alignCenter"
              >
                Clear
              </p>
              <h2
                class="MuiTypography-root MuiTypography-h4 MuiTypography-gutterBottom MuiTypography-alignCenter"
              >
                277°C
              </h2>
              <p
                class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-colorTextSecondary MuiTypography-alignCenter"
              >
                Feels like 
                287°C
              </p>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4 MuiGrid-grid-lg-4"
        >
          <div
            class="MuiPaper-root MuiCard-root card MuiPaper-elevation1 MuiPaper-rounded"
          >
            <div
              class="MuiCardContent-root"
            >
              <h6
                class="MuiTypography-root MuiTypography-subtitle2 MuiTypography-colorTextSecondary MuiTypography-alignCenter"
              >
                24 Apr, 2021
              </h6>
              <img
                alt="forecast-icon"
                class="MuiCardMedia-root image MuiCardMedia-media MuiCardMedia-img"
                src="http://openweathermap.org/img/wn/01n@2x.png"
              />
              <p
                class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-alignCenter"
              >
                Clear
              </p>
              <h2
                class="MuiTypography-root MuiTypography-h4 MuiTypography-gutterBottom MuiTypography-alignCenter"
              >
                276°C
              </h2>
              <p
                class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-colorTextSecondary MuiTypography-alignCenter"
              >
                Feels like 
                276°C
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
